framework/src/actions/ActionWarehouse.C

```diff
@@ -312,6 +309,9 @@ ActionWarehouse::executeAllActions()
 void
 ActionWarehouse::executeActionsWithAction(const std::string & task)
 {
+  // Set the current task name
+  _current_task = task;
```

Comment:
Well it was in order to clean up MooseApp.C where I added the new "uniform_refine_mesh" and had it fire an existing Action manually. There's no place in the code where we hitting this potential bug yet but now it's fixed.